medusa/clients/torrent/transmission_client.py

```diff
@@ -29,6 +32,52 @@
 log.logger.addHandler(logging.NullHandler())
 
 
+class TransmissionBasicAuth(auth.HTTPBasicAuth):
+    """Attaches JWT Bearer Authentication to a TVDB request."""
+
+    refresh_window = 3600  # seconds
+
+    def __init__(self, username, password):
+        """Create a new TVDB request auth."""
+        super(TransmissionBasicAuth, self).__init__(username, password)
+        self.token = None
+        self.token_expires = None
+
+    def login(self, request, data):
+        if self.token:
+            request.headers.update({'x-transmission-session-id': self.token})
+
+        return requests.post(
+            request.url,
+            headers=request.headers,
+            data=data,
+            verify=False,
+            auth=auth.HTTPBasicAuth(self.username, self.password)
+        )
+
+    def authenticate(self, request):
+        """Acquire or refresh a JSON Web Token."""
+        if not self.token or self.token_expires < time():
+            # get token
+            data = {'method': 'session-get'}
+            r = self.login(request, json.dumps(data))
+
+            if r.status_code == 409 and r.headers.get('x-transmission-session-id'):
+                self.token = r.headers.get('x-transmission-session-id')
+                self.token_expires = time() + 3600
+
+            request.headers.update({'x-transmission-session-id': self.token})
+
+    def __call__(self, request):
+        self.authenticate(request)
+        request.headers.update({'x-transmission-session-id': self.token})
+        return super(TransmissionBasicAuth, self).__call__(request)
```

Comment:
If the session id is meant to contain the authentication information, then calling `super().__call__` is incorrect because it would resend the username and password for every request.